oneargopy/Argo.py

```diff
@@ -227,54 +345,432 @@ def __load_sprof_dataframe(self) -> pd:
         prof_index.insert(2, "R_file", R_file)
 
+        # Add profile_index column
+        prof_index.sort_values(by=['wmoid', 'date'], inplace=True)
+        prof_index.insert(0, "profile_index", 0)
+        prof_index['profile_index'] = prof_index.groupby('wmoid')['date'].cumcount() + 1
+
         return prof_index
+    
+
+    def __mark_bgcs_in_prof(self):
+        """ A function to mark weather the floats listed in prof_index are 
+            biogeochemical floats or not. 
+        """
+        bgc_floats = self.sprof_index['wmoid'].unique()
+        is_bgc = self.prof_index['wmoid'].isin(bgc_floats)
+        self.prof_index.insert(1, "is_bgc", is_bgc)
+
+    def __load_is_bgc_index(self)-> pd:
+        """ Function to create a dataframe with float IDs and
+            their is_bcg status for use in select_profiles().
+        """ 
+        float_bgc_status = self.prof_index[['wmoid', 'is_bgc']].drop_duplicates()
+        return float_bgc_status
+
 
     def __display_floats(self) -> None:
-        """ A function to display information about the number of floats.
+        """ A function to display information about the number of floats initially 
+            observed in the unfiltered dataframes. 
         """
         floats = self.prof_index['wmoid'].unique()
         profiles = self.prof_index['file'].unique()
-        print(f"\n{len(floats)} floats with {len(profiles)} profiles found.\n")
+        print(f"\n{len(floats)} floats with {len(profiles)} profiles found.")
 
         bgc_floats = self.sprof_index['wmoid'].unique()
         profiles = self.sprof_index['file'].unique()
-        print(f"{len(bgc_floats)} BGC floats with {len(profiles)} profiles found.\n")
+        print(f"{len(bgc_floats)} BGC floats with {len(profiles)} profiles found.")
+
+
+    def __validate_lon_lat_limits(self)-> None:
+        """ Function to validate the length, order, and contents of longitude and latitude limits passed 
+            to select_profiles.
+        """
+        if self.download_settings.verbose: print(f'Validating longitude and latitude limits...')
+
+        # Validating Lists
+        if len(self.lon_lim) != len(self.lat_lim):
+            raise Exception(f'The length of the longitude and latitude lists must be equal.')
+        if len(self.lon_lim) == 2:
+            if (self.lon_lim[1] <= self.lon_lim[0]) or (self.lat_lim[1] <= self.lat_lim[0]):
+                if self.download_settings.verbose: print(f'Longitude Limits: min={self.lon_lim[0]} max={self.lon_lim[1]}')
+                if self.download_settings.verbose: print(f'Latitude Limits: min={self.lat_lim[0]} max={self.lat_lim[1]}')
+                raise Exception(f'When passing longitude and latitude lists using the [min, max] format the max value must be greater than the min value.')
+            if (abs(self.lon_lim[1]) - self.lon_lim[0] < self.epsilon) and (abs(self.lat_lim[1]) - self.lat_lim[0] < self.epsilon): 
```

Comment:
if (abs(self.lon_lim[1] - self.lon_lim[0] - 360.0) < self.epsilon) and (abs(self.lat_lim[1] - self.lat_lim[0] - 180.0) < self.epsilon):